types/tx.go

```diff
@@ -137,3 +139,10 @@ func TxProofFromProto(pb tmproto.TxProof) (TxProof, error) {
 
 	return pbtp, nil
 }
+
+// ComputeProtoOverheadForTx calculates the size for a protobuf encoded transaction.
+// https://developers.google.com/protocol-buffers/docs/encoding
+func ComputeProtoOverheadForTx(tx Tx) int64 {
```

Comment:
Do you mind tweaking the function name. It no longer calculates the overhead but the entire size of the proto-encoded transaction